### ISSUE TITLE
Intermittent TypeError in Music App: Cannot Read "getIconUrl" Property

### ISSUE THREAD

**Issue by andrewfromtver** [owncloud/music#1137]
**Steps to reproduce**
* Install Music app
* Go to Music tab

**Versions:**

* Nextcloud version: 28.0.3
* Music app version: 1.10.0

**Description:**

The issue occurs intermittently in the Music application where it fails to load properly, resulting in a TypeError. The error message indicates that the application is attempting to read properties of an undefined object, specifically 'getIconUrl'. This issue disrupts the normal functioning of the application.

**Error Log:**

```
webpack.app.ed7a3d4fa79d10ae7b4c.js?v=acdb7557-14:2 TypeError: Cannot read properties of undefined (reading 'getIconUrl')
    at Object.<anonymous> (webpack.app.ed7a3d4fa79d10ae7b4c.js?v=acdb7557-14:2:303745)
    at Object.invoke (webpack.app.ed7a3d4fa79d10ae7b4c.js?v=acdb7557-14:2:60133)
    at webpack.app.ed7a3d4fa79d10ae7b4c.js?v=acdb7557-14:2:94084
    at se (webpack.app.ed7a3d4fa79d10ae7b4c.js?v=acdb7557-14:2:85003)
    at webpack.app.ed7a3d4fa79d10ae7b4c.js?v=acdb7557-14:2:78457
    at se (webpack.app.ed7a3d4fa79d10ae7b4c.js?v=acdb7557-14:2:85649)
    at webpack.app.ed7a3d4fa79d10ae7b4c.js?v=acdb7557-14:2:78457
    at webpack.app.ed7a3d4fa79d10ae7b4c.js?v=acdb7557-14:2:77772
    at webpack.app.ed7a3d4fa79d10ae7b4c.js?v=acdb7557-14:2:44174
    at c.$eval (webpack.app.ed7a3d4fa79d10ae7b4c.js?v=acdb7557-14:2:136349)
(anonymous) @ webpack.app.ed7a3d4fa79d10ae7b4c.js?v=acdb7557-14:2
```
**Screenshot:**

![image](https://github.com/owncloud/music/assets/59865403/67f94577-5576-41fb-b0ee-b74f9198110a)

**Comment by paulijar** [comment]
Thanks for the report. I haven't seen this happen, but the only line of code calling `getIconUrl` is this one here:
https://github.com/owncloud/music/blob/697990b11d40eecd6b4b1ef677de5ea4ae3cea23/js/app/controllers/maincontroller.js#L27

I presume that, on your system, the global object `OC.MimeType` is not always initialized by the time we execute that line. This may depend on many factors, including at least the exact version of Nextcloud, the used browser, and random factors. I guess we need to add some logic there, checking the existence of `OC.MimeType` and scheduling a new check to happen later if the reference is not present yet.

**Comment by paulijar** [comment]
This is now fixed in Music v1.11.0.

**Comment by andrewfromtver** [comment]
Great news, thanks! Updated music app right now and confirm that everything works perfect now!
